cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -67,16 +69,24 @@
   private final SessionManager _sessionManager;
   private final long _maxBlockMs;
   private final ThreadLocal<Integer> _asyncOperationStep;
+  private final Map<EndPoint, Meter> _requestMeter = new HashMap<>();
+  private final Map<EndPoint, Timer> _requestExecutionTimer = new HashMap<>();
 
   public KafkaCruiseControlServlet(AsyncKafkaCruiseControl asynckafkaCruiseControl,
                                    long maxBlockMs,
                                    long sessionExpiryMs,
                                    MetricRegistry dropwizardMetricRegistry) {
     _asyncKafkaCruiseControl = asynckafkaCruiseControl;
-    _sessionManager = new SessionManager(5, sessionExpiryMs, Time.SYSTEM, dropwizardMetricRegistry);
+    _sessionManager = new SessionManager(5, sessionExpiryMs, Time.SYSTEM, dropwizardMetricRegistry, _requestExecutionTimer);
     _maxBlockMs = maxBlockMs;
     _asyncOperationStep = new ThreadLocal<>();
     _asyncOperationStep.set(0);
+
+    for (EndPoint endpoint : EndPoint.values()) {
```

Comment:
Let's add 
1. `private static final List<EndPoint> CACHED_VALUES = Collections.unmodifiableList(Arrays.asList(values()));`, and 
2. the following:
```
  /**
   * Use this instead of values() because values() creates a new array each time.
   * @return enumerated values in the same order as values()
   */
  public static List<Endpoint> cachedValues() {
    return CACHED_VALUES;
  }
```
 to the `Endpoint.java` and call this method rather than the values.